core/src/main/java/com/microsoft/applicationinsights/internal/perfcounter/CpuPerformanceCounterCalculator.java

```diff
@@ -68,4 +71,10 @@ public double getProcessCpuUsage() {
 
         return processCpuUsage;
     }
+
+    private long getProcessCpuTime() throws Exception {
```

Comment:
what is the performance impact of this lookup? Does this change of behavior add more CPU utilization. Upon looking the description of this method it seems like it creates a static object once and then keeps it in the memory so on all subsequent calls it pulls up the same object. https://docs.oracle.com/javase/8/docs/api/java/lang/management/ManagementFactory.html#getPlatformMXBean-java.lang.Class-
Can we instead of calling getPlatformMBeanServer() everytime do it once in the static block/constructor and store it as instance variable in the class. It will save the compiler time of calling the method.

Reply:
I ran some basic tests and the performance looks to be the same as before this change.

I like how @dan-crough-bmi refactored this and I'm going to integrate their changes. Thanks for contributing, @dan-crough-bmi !